packages/lit-html/src/lib/lit-html.ts

```diff
@@ -189,14 +189,14 @@ class Template {
     // we also dynamically create a regex to find the matching end tags for raw
     // text elements.
 
-    const textRegex = /<((?:!--)|(?:\w*))/g;
+    const textRegex = /<(!--|\/?\w+|(\W|$))/g;
```

Comment:
Based on that the dynamic tag comment, I think this should be:
```suggestion
    const textRegex = /<(!--|\/?\w+|(\/?$))/g;
```

Or, we could go even further with `/<(!--|\/?(\w+|$))/g`

Reply:
I think it's somewhat similar in spirit to this, except that I keep each next state of _comment_ or _tag_ or _<_ or _dynamic_tag_ in a separate group to make the logic for dealing with the matches easier. There's probably room for optimization, but I want to get the attribute tests in first.